packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -323,10 +324,10 @@ export const usePickFiles: EnsembleActionHook<PickFilesAction> = (
 
   useEffect(() => {
     if (!isEmpty(values?.files) && !isComplete) {
-      onCompleteAction?.callback({ files: values?.files });
+      onCompleteAction?.callback({ files: values?.files, ...scope });
```

Comment:
What action are you trying to invoke here that is missing the scope?

Reply:
i am trying to access the item-template `name` under `pickFiles` to modify the the data

Reply:
Can you share the YAML block? The `pickFiles` action should have access to the custom scope via `useRegisterBindings`

Reply:
yes here 
```yaml
Button:
                              label: Upload File
                              onTap:
                                pickFiles:
                                  id: ${"uploadDoc"+docList.index}
                                  allowMultiple: true
                                  allowedExtensions:
                                    - pdf
                                    - docx
                                  onComplete:
                                    executeCode: |
                                      const fileName = files[0].name;
                                      const date = formatDate(files[0].lastModifiedDate);
                                      const data = ensemble.storage.get('docsData');
                                      const newData = data.map((d) => {
                                        if(d.index === docList.index){
                                          d.fileName = fileName;
                                          d.date = date;
                                        }
                                        return d;
                                      });
                                      ensemble.storage.set('docsData', newData);
```

Reply:
So the ID is the issue? For executeCode, we would need to add the custom scope hook there.

Reply:
so are we gonna approve it @evshi ?